src/openforms/js/components/admin/forms/VariableSelection.js

```diff
@@ -24,26 +25,51 @@ const VariableSelection = ({
   });
 
   const allFormVariables = (includeStaticVariables ? staticVariables : []).concat(formVariables);
+
+  const getVariableSource = variable => {
+    if (variable.source === VARIABLE_SOURCES.userDefined) {
+      return VARIABLE_SOURCES_GROUP_LABELS.userDefined;
+    }
+    if (variable.source === VARIABLE_SOURCES.component) {
+      return VARIABLE_SOURCES_GROUP_LABELS.component;
+    }
+    return VARIABLE_SOURCES_GROUP_LABELS.static;
+  };
+
   const choices = allFormVariables
     .filter(variable => filter(variable))
-    .map(variable => {
-      const label = formDefinitionsNames[variable.formDefinition]
-        ? `${formDefinitionsNames[variable.formDefinition]}: ${variable.name} (${variable.key})`
-        : `${variable.name} (${variable.key})`;
-      return [variable.key, label];
-    });
-
-  {
-    /*TODO: This should be a searchable select for when there are a billion variables -> react-select */
-  }
+    .reduce(
+      (variableGroups, variable) => {
+        let label = `<span class="form-variable-dropdown__option__label">${variable.name} <code class="form-variable-dropdown__option__key">(${variable.key})</code></span>`;
+        if (formDefinitionsNames[variable.formDefinition]) {
+          label += `<span class="form-variable-dropdown__option__form-definition">${formDefinitionsNames[variable.formDefinition]}</span>`;
+        }
```

Comment:
you should be able to just return JSX instead of HTML strings for the label. `dangerouslySetInnerHTML` should be avoided at all costs